### PR TITLE
ref(js): Fix incorrect interpolation in t()

### PR DESCRIPTION
**Pull request by evanpurkhiser**
The `t` function should ONLY accept static strings. Since the way
translations work is that we extract out the static string *at compile
time* (that is, when webpack is run with SENTRY_EXTRACT_TRANSLATIONS)
and use those strings as the keys for the translator.

 [!!] This means these strings CANNOT be dynamic! The translated string
      will not be able to be looked up if it is not a static string!

 Because of this interpolation MUST be done by using the
 sprintf style parameterization

### CODE REVIEW

**Review by priscilawebdev — APPROVED** [review]
good catch! Thanks for fixing it 👏

**Comment by evanpurkhiser** [comment]
Created an eslint rune to stop us from doing this 💯 https://github.com/getsentry/eslint-config-sentry/pull/160

File: static/app/components/organizations/pageFilters/desyncedFiltersAlert.tsx [hunk 1]
```diff
@@ -76,14 +76,14 @@ export default function DesyncedFilterAlert({
         </Fragment>
       }
     >
-      {tct(
-        message ??
+      {message ??
```
Comment (evanpurkhiser) [inline static/app/components/organizations/pageFilters/desyncedFiltersAlert.tsx:79]: I checked that message wasn't inserting any of the `filter` or `has` parameters.

File: static/app/components/organizations/pageFilters/desyncedFiltersAlert.tsx [hunk 2]
```diff
@@ -76,14 +76,14 @@ export default function DesyncedFilterAlert({
...
+        tct(
           'The [filter] [has] been overwritten. This can happen when you open sentry.io links with filter parameters that are different from your current filter values.',
-        {
-          filter: getReadableDesyncedFilterList(desyncedFilters),
-          has: desyncedFilters.size > 1 ? t('have') : t('has'),
-        }
-      )}
+          {
+            filter: getReadableDesyncedFilterList(desyncedFilters),
+            has: desyncedFilters.size > 1 ? t('have') : t('has'),
```
Comment (evanpurkhiser) [inline static/app/components/organizations/pageFilters/desyncedFiltersAlert.tsx:84]: uhg, this actually needs to be a `tn`

**Review by priscilawebdev — APPROVED** [review]
Jesus 😲

File: static/app/views/issueList/overview.tsx [hunk 3]
```diff
@@ -1085,13 +1089,13 @@ class IssueListOverview extends Component<Props, State> {
     actionType: 'Reviewed' | 'Resolved' | 'Ignored'
   ) => {
     if (itemIds.length > 1) {
-      addMessage(t(`${actionType} ${itemIds.length} Issues`), 'success', {
+      addMessage(`${actionType} ${itemIds.length} ${t('Issues')}`, 'success', {
```
Comment (evanpurkhiser) [inline static/app/views/issueList/overview.tsx:1092]: Kinda gross, but there's not really enough context here

File: static/app/components/organizations/pageFilters/desyncedFiltersAlert.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/components/organizations/pageFilters/desyncedFiltersAlert.tsx:84]: yes ☝️

File: static/app/views/issueList/overview.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/views/issueList/overview.tsx:1092]: agree..but maybe we can refactor it in a follow up

Comment (evanpurkhiser) [reply static/app/views/issueList/overview.tsx:1092]: 👍

**Pull request merged**